### ISSUE TITLE
Ablation Studies:  RiRoI Align

### ISSUE THREAD

**Issue by chandlerbing65nm** [csuhan/ReDet#94]
Hello, Good day! I hope you are doing well.

I tried recreating the ablation studies you performed in the paper with DOTA v1.5. 

Specifically, I retrain ReDet using the codes below to verify if the RiRoI align really helps in detecting oriented objects.

1. w/ RiRoI: `roi_layer=dict(type='RoIAlignRotated', out_size=7, sample_num=2),`
2. w/o RiRoI: `roi_layer=dict(type='RiRoIAlign', out_size=7, sample_num=2),` 

But I get:
- w/ RiRoI: **0.6677 mAP**
- w/o RiRoI: **0.6695 mAP**

In the published paper, I should get a **+0.85 mAP** increase.

I used 1 GPU: NVIDIA V100 and a learning rate of 0.0025. Other parameters are the same with your implementation.

**Comment by csuhan** [comment]
Good question!
First, I have to admit your result is reasonable (even our results with RiRoIAlign have not never > 66.0%). While, our ablation results are observed by multiple runs and across multiple datasets.
Here I list some suggestions that may help you.
1. DOTA is not as stable as we think. We choose DOTA-v1.5 other than DOTA-v1.0 because DOTA-v1.5 is annotated with more small objects. From this view, I suggest you to check class-wise results, and see if there is a shift (in class AP) between RRoIAlign and RiRoIAlign. Also, multiple runs may show you the real performance.
2. We observed the performace is different with different number of GPU. Generally, 4 GPUs performs better than 8 GPUs. But I have not tested on 1 GPU.

**Comment by chandlerbing65nm** [comment]
@csuhan Thank you very much for the suggestions. I appreciate it a lot.

I'm currently running the ablation experiment on the equivariant backbone. I'm using baseline Faster RCNN w/ FPN to place and remove these codes:

1. **w/ equivariant backbone** (ReR50-ReFPN)
```
model = dict(
    type='FasterRCNN',
    pretrained='work_dirs/ReResNet_pretrain/re_resnet50_c8_batch256-25b16846.pth',
    backbone=dict(
        type='ReResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        style='pytorch'),
    neck=dict(
        type='ReFPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        num_outs=5),
```

2. **w/o equivariant backbone** (R50-FPN)
```
model = dict(
    type='FasterRCNN',
    pretrained='modelzoo://resnet50',
    backbone=dict(
        type='ResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        style='pytorch'),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        num_outs=5),
```
From the published paper, I should expect to have a very big improvement in the model size. 

Is this how you conducted the ablation experiment on the equivariant backbone?

**Comment by csuhan** [comment]
Yes. The only difference is their `backbone` and `neck`.

**Comment by chandlerbing65nm** [comment]
@csuhan Hello again. I just want to ask, how did you calculate the model size for _FasterRCNNOBB with equivariant backbone_ (ReR50-ReFPN)? 

I have finished the **_epoch_12.pth_** training, the size when I view the properties is **_139 Mb_**.

**Comment by csuhan** [comment]
https://github.com/csuhan/ReDet/blob/master/tools/counting_ReDet_model_size.py

**Comment by chandlerbing65nm** [comment]
Thank you very much!
